gridstatus/ercot.py

```diff
@@ -3351,3 +3353,56 @@ def _weather_zone_column_name_order(self):
             "Southern",
             "West",
         ]
+
+    @support_date_range(frequency=None)
+    def get_indicative_lmp_by_settlement_point(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        if date == "latest":
+            self.get_indicative_lmp_by_settlement_point(date="today")
+        if not end:
+            end = date + pd.DateOffset(days=1)
+
+        docs = self._get_documents(
+            report_type_id=ERCOT_INDICATIVE_LMP_BY_SETTLEMENT_POINT_RTID,
+            date=date,
+            extension="csv",
+            published_before=end,
+            published_after=date,
+            verbose=verbose,
+        )
+        df = self.read_docs(docs, parse=False, verbose=verbose)
+        columns_to_rename = {
+            "RepeatedHourFlag": "DSTFlag",
+            "IntervalId": "Interval Id",
+            "IntervalEnding": "Interval End",
+            "IntervalRepeatedHourFlag": "Interval Repeated Hour Flag",
+            "SettlementPoint": "Location",
+            "SettlementPointType": "Location Type",
+            "LMP": "LMP",
+        }
+        df.rename(columns=columns_to_rename, inplace=True)
+        df["RTDTimestamp"] = pd.to_datetime(df["RTDTimestamp"]).dt.tz_localize(
```

Comment:
Got it. I think I've implemented this logic correctly, though I kind of copy it into the `_handle_indicative_lmp_by_settlement_point` rather than try to use/refactor `parse_doc` for now. 

Still running some tests, but the fall back hours seems to be working in the TestErcotAPI test, and the spring forward date (full day of 2024-03-10) seems to work when I pull data with the getter method. 

But trying to have the test just test the hours from 1am to 4am seems to fail a bit still.

Reply:
is this exactly copy and pasted? if so, i'd this we can figure out how to refactor safely. im okay if we want to leave it as a todo, but then let's add the todo comment